### PR TITLE
Added in support for the new api method call for /doc_logs which allo…

### PR DESCRIPTION
**Pull request by jbrodie**
…ws searching based on time frame

http://docraptor.com/documentation#doc_log_listing

### CODE REVIEW

File: test/doc_logs_bang_test.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    describe "with bogus arguments" do
+      it "should raise an error if something other than an options hash is passed in" do
+        assert_raises(ArgumentError) {DocRaptor.doc_logs!(true)}
+        assert_raises(ArgumentError) {DocRaptor.doc_logs!(nil)}
+      end
+    end
+
+    describe "with good arguments" do
+      it "should give me a valid response" do
+        stub_http_response_with("simple_list_docs.xml", :get)
```
Comment (janxious) [inline test/doc_logs_bang_test.rb]: probably need to add a new fixture that is like 'simple_list_doc_logs.xml' for this case.

**Comment by janxious** [comment]
This looks really solid at a first glance. Need to kick the tires a bit, but probably merge and release in the next 24h if everything goes well. :smile:

File: test/doc_logs_bang_test.rb (hunk 1 above)
Comment (jbrodie) [inline test/doc_logs_bang_test.rb]: I don't believe the return is any different but I can confirm this with Tyler at DocRaptor later today.

Sent from my iPhone

On Aug 5, 2015, at 10:42 AM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/doc_logs_bang_test.rb<https://github.com/expectedbehavior/doc_raptor_gem/pull/14#discussion_r36309152>:

> +class DocLogsBangTest < MiniTest::Test
> +  describe "doc_logs!" do
> +    before do
> +      DocRaptor.api_key "something something"
> +    end
> +
> +    describe "with bogus arguments" do
> +      it "should raise an error if something other than an options hash is passed in" do
> +        assert_raises(ArgumentError) {DocRaptor.doc_logs!(true)}
> +        assert_raises(ArgumentError) {DocRaptor.doc_logs!(nil)}
> +      end
> +    end
> +
> +    describe "with good arguments" do
> +      it "should give me a valid response" do
> +        stub_http_response_with("simple_list_docs.xml", :get)


probably need to add a new fixture that is like 'simple_list_doc_logs.xml' for this case.

**Comment by jbrodie** [comment]
Excellent. Thanks.

Sent from my iPhone

On Aug 5, 2015, at 10:43 AM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


This looks really solid at a first glance. Need to kick the tires a bit, but probably merge and release in the next 24h if everything goes well. [https://assets-cdn.github.com/images/icons/emoji/unicode/1f604.png]

**Comment by janxious** [comment]
This is merged now, and I did a little bit of work to get it working against production endpoints, as well as finding some places where DR itself needs some updates. You can probably roll with master if you need doc logs *right now*, and will get a new gem release out soonish.

**Comment by jbrodie** [comment]
Sooner you can post that gem the better.

I am building a rails mountable engine and can not reference git in the gemspec.

If you could increment the gem and roll it out that would be most excellent. I have all the UI done for the engine but not the sync yet.

I also sent Tyler an email about some of the fields. The call isn't returning the doc raptor log id which I would like to have for uniqueness. There were also a couple other questions I had on the return output.

Jay


Sent from my iPhone

On Aug 7, 2015, at 1:57 PM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


This is merged now, and I did a little bit of work to get it working against production endpoints, as well as finding some places where DR itself needs some updates. You can probably roll with master if you need doc logs right now, and will get a new gem release out soonish.

**Comment by janxious** [comment]
Cool. Checking out that email and doc log id adding will be pretty simple. The others are just going to take time to iterate through. :smile_cat:

**Comment by jbrodie** [comment]
Ya, the docs show a bunch of prince options but I have no idea of the format they are returned in the doc log call.


Sent from my iPhone

On Aug 7, 2015, at 2:07 PM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


Cool. Checking out that email and doc log id adding will be pretty simple. The others are just going to take time to iterate through. [https://assets-cdn.github.com/images/icons/emoji/unicode/1f638.png]

**Comment by janxious** [comment]
@jbrodie The API should now return doc log ids are part of its payload.

**Comment by jbrodie** [comment]
Excellent.
Thanks. I will check it out.


Sent from my iPhone

On Aug 7, 2015, at 3:17 PM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


@jbrodie<https://github.com/jbrodie> The API should now return doc log ids are part of its payload.

**Comment by janxious** [comment]
@jbrodie New version (0.5.0) is up: https://rubygems.org/gems/doc_raptor

Note that I changed method names on you for the doc log listing endpoint.

**Comment by jbrodie** [comment]
Awesome. Thanks for the heads up, is the new method name updated on gem read me?  I can check for it there.


Sent from my iPhone

On Aug 7, 2015, at 3:28 PM, janxious <notifications@github.com<mailto:notifications@github.com>> wrote:


@jbrodie<https://github.com/jbrodie> New version (0.5.0) is up: https://rubygems.org/gems/doc_raptor

Note that I changed method names on you for the doc log listing endpoint.

**Comment by janxious** [comment]
https://github.com/expectedbehavior/doc_raptor_gem/blob/master/Changelog.md

tl;dr: `list_doc_logs`

**Comment by jbrodie** [comment]
So working through setting up the sync stuff, how can I pull a list of documents, and get the total count of pages left?  I don't see anything out there to make this work.

So here s the scenario, I am going to be fetching them nightly, so it shouldn't be a huge thing, but when I pull doc logs based on a date range, I have to give a page number to fetch from and a total count.  It doesn't look like the response returns any data associated with the number of pages remaining, or even the next page number if there is one.

The only work around I can do on this is check to make sure that the number I was expecting to be returned was, and then increment the page number by one and pull again, until I get an empty result.  This isn't ideal by any means but building pagination into your api isn't a short order.  I will build a work around for now but would love to see this properly structured through the API at some point in time to make it proper.


Jay Brodie
Senior Developer
Info-Tech Research Group

**Comment by janxious** [comment]
I thought there was a high probability you would run into this. I'm not sure what that will look like in our system, but I certainly think it would make the API better. My suggestion for current workaround was to do exactly as you described wherein you request next page until the request no longer returns results.

**Comment by jbrodie** [comment]
Ya I am just working around it right now in that fashion.

I should have something done by Monday that I can show you to see what ya think!

Should be a simple setup for any rails user.

Jay Brodie
Senior Developer
Info-Tech Research Group
